modules/swagger-codegen/src/main/resources/python/api_client.mustache

```diff
@@ -625,3 +640,124 @@ class ApiClient(object):
+        """
+        This method prepares the Auth header string
+        
+        :param hdrs : HTTP Headers
+        :param signed_msg: Signed Digest
+        :return: instance of digest object 
+        """
+        
+        auth_str = ""
+        auth_str = auth_str + "Signature"
+    
+        auth_str = auth_str + " " + "keyId=\"" + auth_settings['api_key_id'] + "\"," + "algorithm=\"" + auth_settings['digest_algorithm'] + "\"," + "headers=\"(request-target)"
+    
+        for key, _ in hdrs.items():
+            auth_str = auth_str + " " + key.lower()
+        auth_str = auth_str + "\""
+    
+        auth_str = auth_str + "," + "signature=\"" + signed_msg.decode('ascii') + "\""
+    
+        return auth_str
+    
+    def prepare_auth_header(self, auth_settings, resource_path, method, body, query_params):
+        """
+        Method to perform operations required to prepare the auth header and eventually
+        add the signed headers to default_header object used by ApiClient class.
+        
+        https://datatracker.ietf.org/doc/draft-cavage-http-signatures/?include_text=1
+        
+        :param auth_setting: Authsettings dictionary
+        :param resource_path : resource path which is the api being called upon.
+        :param method: request type
+        :param body: body passed in the http request.
+        :param query_params: query parameters used by the API 
+        :return: instance of digest object 
+        """
+        if body is None:
+            body = ''
+        else:
+            body = json.dumps(body)
+        
+        
+        target_host = urlparse(self.host).netloc
+        target_path = urlparse(self.host).path
+        
+        request_target = method + " " + target_path + resource_path
+        
+        if query_params:
+            raw_query = urlencode(query_params).replace('+', '%20')
+            request_target += "?" + raw_query
+        
+        from email.utils import formatdate
+        cdate=formatdate(timeval=None, localtime=False, usegmt=True)
+        
+        request_body = body.encode()
+        body_digest = self.get_sha256_digest(request_body)
+        b64_body_digest = b64encode(body_digest.digest())
+        
+        headers = {'Content-Type': 'application/json', 'Date' : cdate, 'Host' : target_host, 'Digest' : "SHA-256=" + b64_body_digest.decode('ascii')}
+        
+        string_to_sign = self.prepare_str_to_sign(request_target, headers)
+        
+        if auth_settings['digest_algorithm'] != 'rsa-sha256':
+                raise ValueError('only rsa-sha256 supported for digest_algorithm in httpsig')
+        digest = self.get_sha256_digest(string_to_sign.encode())
+        
+        b64_signed_msg = self.get_rsasig_b64encode(auth_settings['private_key_file'], digest)
+        
+        auth_header = self.get_auth_header(auth_settings, headers, b64_signed_msg)
+        
+        self.set_default_header('Date', '{0}'.format(cdate))
```

Comment:
If you only need to make a string from it, just use `str` instead of `'{0}'.format`